selfdrive/car/hyundai/values.py

```diff
@@ -392,7 +394,8 @@ def get_platform_codes(fw_versions: List[bytes]) -> Set[bytes]:
 
 # Regex patterns for parsing platform code, FW date, and part number from FW versions
 PLATFORM_CODE_FW_PATTERN = re.compile(b'((?<=' + HYUNDAI_VERSION_REQUEST_LONG[1:] +
-                                      b')[A-Z]{2}[A-Za-z0-9]{0,2})(?:.*([0-9]{6}))?')
+                                      b')[A-Z]{2}[A-Za-z0-9]{0,2})')
+DATE_FW_PATTERN = re.compile(b'(?<=[ -])([0-9]{6}$)')
 PART_NUMBER_FW_PATTERN = re.compile(b'(?<=[0-9][.,][0-9]{2} )([0-9]{5}[-/]?[A-Z][A-Z0-9]{3}[0-9])')
```

Comment:
Now checks that date string is at the end of the FW version as well as is preceded by a space or dash (Hyundai Genesis coming in yet again with the exception)